src/witan/sen2/return/person_level/blade/plans_placements.clj

```diff
@@ -341,154 +331,241 @@
 
 
 ;;; # Checks
-(def checks
-  "Definitions for standard checks for issues in dataset of plans & placements on census dates."
+(defn checks
+  "Definitions of checks for issues in dataset of plans & placements on census dates."
+  [& {:keys [sen-settings sen-types]
+      :or   {sen-settings sen2-dictionary/sen-settings
+             sen-types    sen2-dictionary/sen-types}}]
   (let [count-true? #(comp count (partial filter true?) %)
         ;; Check definitions:
-        ;; - `:col-fn`s:
-        ;;   - are applied to the dataset in turn.
-        ;;   - should (only) add a single column to the dataset, with name matching the check key.
-        ;;   - should otherwise leave the dataset intact.
-        m           {:issue-non-unique-key
+        ;; - `:col-fn`s: check function supplied to tc/add-column so receives dataset as single argument and
+        ;;               must return either a column, sequence or single value with truthy values indicating an issue.
+        m           {:issue-missing-person-table-id
                      {:idx           001
+                      :label         "Missing :person-table-id"
+                      :cols-required #{:person-table-id}
+                      :col-fn        #(->> % :person-table-id (map nil?))
+                      :summary-fn    (count-true? :issue-missing-person-table-id)
+                      :summary-label "#rows"
+                      :action        "Required: fill in the blanks."}
+                     :issue-missing-census-year
+                     {:idx           002
```

Comment:
why are you creating octal literals?

Reply:
haha, no, was using leading digits to separate checks on different aspects of the plans & placements and never got to 9.